packages/angular_devkit/build_angular/src/utils/index-file/augment-index-html.ts

```diff
@@ -40,6 +40,7 @@ export interface AugmentIndexHtmlOptions {
   /** Used to set the document default locale */
   lang?: string;
   hints?: { url: string; mode: string; as?: string }[];
+  imageDomains?: string[];
```

Comment:
You should be able to use the `hints` option to add the preconnects:
``
hints: [{ mode: 'preconnect', url: 'https://example.com' }, ...],
``

Reply:
There's a little extra logic for image domains beyond just generic preconnects. I just pushed a change to this PR where we're [checking the image domains vs any preconnects](https://github.com/angular/angular-cli/pull/25709/files#diff-c8f27b3539a3f2924e95325c9d6105de9f37ff3b73cbcd7ca15223d461335e43R238) that the end user has manually included in the index.html file, to make sure not to add duplicates. That would be harder to do if the image domains were just mixed in with the rest of the preconnect hints. Do you think it's still worth trying to avoid using the extra data sctructure?

Reply:
Cant the hints themselves be deduplicated in a generic way?